Resources/Private/Javascript/Connector.js

```diff
@@ -83,23 +84,47 @@ export class ShopwareConnector {
     });
   }
 
-  addProductToCart(productId) {
+  async addProductToCart(productId) {
+    fireEvent('begin_add-to-cart');
+
     const url = endpoints.ADD_TO_CART_BASE + productId;
-    return this.request('post', url);
```

Comment:
Not returning a Promise here means not being able to handle async code in your own app, solely relying on events is not a good idea and you want to give developers the freedom to choose their weapon. Especially with this implementation, which forces you handle every error-Event in every component. Normally you want to handle specific errors in the specified component and not have an overall error-Listener that has to check the error message before handling the error or not